basket/news/backends/braze.py

```diff
@@ -0,0 +1,181 @@
+        @raises: BrazeRateLimitError: 429 error (rate limit exceeded)
+        @raises: BrazeInternalServerError: 500 error (Braze server error)
+        @raises: BrazeClientError: any other error
+
+        """
+        if not self.active:
+            return
+
+        url = urljoin(self.api_url, endpoint.value)
+
+        headers = {
+            "Authorization": f"Bearer {self.api_key}",
+            "Content-Type": "application/json",
+        }
+
+        try:
+            response = requests.post(url, headers=headers, data=json.dumps(data))
+            response.raise_for_status()
+            return response.json()
+        except requests.exceptions.HTTPError as exc:
+            status_code = exc.response.status_code
+
+            if status_code == 400:
+                raise BrazeBadRequestError from exc
+
+            if status_code == 401:
+                raise BrazeUnauthorizedError from exc
+
+            if status_code == 403:
+                raise BrazeForbiddenError from exc
+
+            if status_code == 404:
+                raise BrazeNotFoundError from exc
+
+            if status_code == 429:
+                raise BrazeRateLimitError from exc
+
+            if status_code >= 500 and status_code <= 599:
+                raise BrazeInternalServerError from exc
+
+            raise BrazeClientError from exc
+
+    def track_user(self, email):
+        """
+        Track a user in Braze.
+
+        In this case, we are creating an alias-only user to send a transactional-like email.
+
+        You can use the /users/track endpoint to create a new alias-only user by setting the
+        _update_existing_only key with a value of false in the body of the request.
+
+        https://www.braze.com/docs/api/endpoints/user_data/post_user_track/
+
+        Note: `email_subscribed` values are:
+          - “opted_in” - explicitly registered to receive email messages
+          - “unsubscribed” - explicitly opted out of email messages
+          - “subscribed” - neither opted in nor out
+
+        """
+        data = {
+            "attributes": [
+                {
+                    "_update_existing_only": False,
+                    "user_alias": {"alias_name": email, "alias_label": "email"},
+                    "email": email,
+                    "email_subscribe": "subscribed",
+                }
+            ],
+        }
+
+        return self._request(BrazeEndpoint.USERS_TRACK, data)
+
+    def export_users(self, email):
+        """
+        Export user profile by identifier.
+
+        https://www.braze.com/docs/api/endpoints/export/user_data/post_users_identifier/
+
+        If alias is not found, returns empty "users" list.
+
+        """
+        data = {
+            "user_aliases": [{"alias_name": email, "alias_label": "email"}],
+            "email_address": email,
+        }
+
+        return self._request(BrazeEndpoint.USERS_EXPORT_IDS, data)
+
+    def send_campaign(self, email, campaign_id):
```

Comment:
Question, non-blocking: do we need a bulk version of this as well?

Reply:
These are just transactional emails even though we are using the campaign API, so shouldn't be called in bulk. Kinda confusing, but apparently transactional emails are an add-on.